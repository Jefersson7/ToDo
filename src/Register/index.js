import React, { Component } from "react"
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  AsyncStorage,
  Alert
} from "react-native"
import t from "tcomb-form-native"

const Form = t.form.Form

class RegisterScreen extends Component {
  constructor(props) {
    super(props)
  }

  options = {
    fields: {
      email: {
        label: "E-Mail"
      },
      password: {
        label: "ContraseÃ±a",
        password: true,
        secureTextEntry: true
      }
    }
  }

  User = t.struct({
    email: t.String,
    password: t.String
  })

  goToLogIn = () => {
    this.props.navigation.navigate("Auth")
  }

  handleSubmit = () => {
    let value = this._form.getValue()
    AsyncStorage.setItem(`${value.email}`, value.password)
    Alert.alert(
      "Registro Exitoso",
      `El usuario con el e-mail ${value.email} se ha registrado exitosamente`,
      [
        {
          text: "Ir al inicio de sesion",
          onPress: () => this.props.navigation.navigate("Auth")
        }
      ]
    )
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.textHeader}>Registarse</Text>
        <Form
          ref={c => (this._form = c)}
          type={this.User}
          options={this.options}
        />
        <View style={styles.options}>
          <TouchableOpacity style={styles.button} onPress={this.handleSubmit}>
            <Text style={styles.buttonText}>Registrarse</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.button} onPress={this.goToLogIn}>
            <Text style={styles.buttonText}>Log In</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 60,
    backgroundColor: "#b8d8d8",
    justifyContent: "center"
  },
  options: {
    textAlign: "center",
    flexDirection: "row",
    justifyContent: "center"
  },
  button: {
    padding: 10,
    marginTop: 20,
    marginLeft: 20,
    marginRight: 20,
    backgroundColor: "#36593b"
  },
  buttonText: {
    fontSize: 17,
    color: "white"
  },
  textHeader: {
    color: "black",
    textAlign: "center",
    fontWeight: "bold",
    marginBottom: 30,
    fontSize: 25
  }
})

export default RegisterScreen
