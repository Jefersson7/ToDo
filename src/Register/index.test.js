import React from "react"
import { shallow, mount } from "enzyme"
import Register from "./index"

describe("Register", () => {
  describe("Rendering", () => {
    it("Should match to snapshot", () => {
      const register = shallow(<Register />)
      expect(register).toMatchSnapshot()
    })
  })
})
