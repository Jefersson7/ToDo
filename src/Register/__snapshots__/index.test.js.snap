// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Rendering Should match to snapshot 1`] = `ShallowWrapper {}`;
